ThreeBotPackages/tfgrid/threebot_provisioning/chatflows/threebot_reservation.py

```diff
@@ -0,0 +1,135 @@
+from Jumpscale import j
+import netaddr
+import random
+import nacl
+import os
+
+
+def chat(bot):
+
+    """
+    This chat is create 3bot container using zosv2
+    """
+    explorer = j.clients.threebot.explorer
+    cl = j.clients.s3.get("deployer")
+
+    AWS_ID = cl.accesskey_
+    AWS_SECRET = cl.secretkey_
+
+    user_info = bot.user_info()
+    name = user_info["username"]
+    email = user_info["email"]
+    ips = ["IPV6", "IPV4"]
+    choose = ["New", "Restore"]
+    ip_range_choose = ["Specify IP Range", "Choose IP Range for me"]
+
+    folder_name = name.replace(".", "_")
+    env = dict()
+    secret_env = dict()
+
+    if not name or not email:
+        bot.md_show("Username or email not found in session. Please log in properly")
+
+    user_choice = bot.single_choice("create a new 3bot or restore existing 3bot?", choose)
+    identity = explorer.actors_all.phonebook.get(name=name, email=email)
+    identity_pubkey = identity.pubkey
+    if user_choice == "Restore":
+        password = bot.secret_ask("please add restore password?")
```

Comment:
and actually, perhaps it would make sense before asking this question is to check first whether you have a backup for the logged in 3bot name....

if not, show an error saying

"The 3bot logged in does not have any backups"

Reply:
Make sense,  should first check in s3 this 3bot have backup directory or not.
this is need some logic to implement .